### PR TITLE
fix(Tooltip): Forward ref to allow nesting

### LINKED ISSUE #299: Tooltip component should be a forwardRef component
#### Describe the bug (actual behavior)
Tooltips isn't a forwardRef component and cannot be add in a DropdownMenu. If added, tooltip work, but no dropdown. 
```
<DropdownMenu menuItems={items}>
  <Tooltip content={"text"}>
    <IconButton>
      <SvgSomething/>
    </IconButton>
  </Tooltip>
</DropdownMenu>
```
`Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?`

Can work with this workflow but unnecessary div wrapper is needed : 
```
<DropdownMenu menuItems={items}>
  <div>
    <Tooltip content={"text"}>
      <IconButton>
        <SvgSomething/>
      </IconButton>
    </Tooltip>
  </div>
</DropdownMenu>
```

#### Expected behavior
No wrapper div is necessary in this case. Seem like Tooltip was design to allow ref as props. Maybe this package could help : react-merge-refs ?

#### Suggestion : 
Having some props of tooltip in Button component ?

### PR DESCRIPTION
**Pull request by mayank99**
Closes #299

## Checklist

- [x] Add meaningful unit tests for your component (verify that all lines are covered)
- [x] Verify that all existing tests pass
- [x] ~~Add component features demo in Storybook (different stories)~~
- [x] ~~Approve test images for new stories~~
- [x] ~~Add screenshots of the key elements of the component~~

### CODE REVIEW

File: src/core/Tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -33,17 +33,8 @@ export type TooltipProps = {
...
-    content,
-    children,
-    className,
-    style,
-    visible,
-    ref,
-    id,
-    ...rest
-  } = props;
+export const Tooltip = React.forwardRef<Element, TooltipProps>((props, ref) => {
```
Comment (veekeys) [inline src/core/Tooltip/Tooltip.tsx:36]: We always used `HTMLElement` as type. Any reason you chose Element here?

File: src/core/Tooltip/Tooltip.tsx [hunk 2]
```diff
@@ -33,17 +33,8 @@ export type TooltipProps = {
...
+  const { content, children, className, style, visible, id, ...rest } = props;
```
Comment (veekeys) [inline src/core/Tooltip/Tooltip.tsx:37]: So even though we pass ref on the `Popover` component, at the end it appears on children? (according to your test)
Then I am lost a little, whats the point of this ref if user can add ref on his provided children himself...

File: src/core/Tooltip/Tooltip.tsx (hunk 1 above)
Comment (mayank99) [reply src/core/Tooltip/Tooltip.tsx:36]: I kept the same type as the one from tippy props

File: src/core/Tooltip/Tooltip.tsx (hunk 2 above)
Comment (mayank99) [reply src/core/Tooltip/Tooltip.tsx:37]: I was a little surprised too, that it appears on children.

I think the point is to allow nesting tippys. See linked issue and also see #296

Comment (veekeys) [reply src/core/Tooltip/Tooltip.tsx:37]: So now the scenario mentioned is working well? without the need of div?

Comment (mayank99) [reply src/core/Tooltip/Tooltip.tsx:37]: Yes, I tested this
image
image

File: src/core/Tooltip/Tooltip.tsx (hunk 1 above)
Comment (veekeys) [reply src/core/Tooltip/Tooltip.tsx:36]: Why I ask, cause usually ppl create refs of `HTMLElement` and then access certain props, which `Element` might not have.. And I wildly guess, that if you create `HTMLElement` ref it wont allow you to set it on Tooltip then..

Comment (mayank99) [reply src/core/Tooltip/Tooltip.tsx:36]: But it won't let me use HTMLElement
image

File: src/core/Tooltip/Tooltip.tsx [hunk 3]
```diff
@@ -68,6 +59,6 @@ export const Tooltip = (props: TooltipProps) => {
       {children && React.cloneElement(children, { title: undefined })}
     </Popover>
```
Comment (maxime4000) [inline src/core/Tooltip/Tooltip.tsx:60]: Does adding the ref in reference={ref} instead of ref in popover make it work ?

Comment (mayank99) [reply src/core/Tooltip/Tooltip.tsx:60]: That's not how `reference` is supposed to be used

**Pull request closed without merging**